### ISSUE TITLE
swaymsg [identifier] move position x y behaves relatively instead of absolute

### ISSUE THREAD

**Issue by LuyeT** [swaywm/sway#5691]
- **Description:**
  - Please describe in plain English how Sway and i3's behaviors differ or what functionality Sway is lacking.
I'm having an issue with the "move position" command in sway.

In i3, executing `i3-msg "[class=\"conky_cpu\"] move position 3000 1059"` will position the conky window 3000 pixels from the left edge of my screenspace, regardless of where the current focus is. In sway however, this appears to be relative to the screen containing currently focused window. I.e. `swaymsg "[class=\"conky_cpu\"] move position 3000 1059"` will move the window outside of a viewable range if executed from a terminal on my rightmost screen.

I switched from i3 to sway just yesterday, so forgive me if my understanding of wayland is lackluster. Nonetheless, I believe the difference in behaviour stems from how wayland sees the screenspace compared to X11. If I'm correct in assuming this stems from a fundamental difference in how X11 and wayland functions, it might not be possible to have this function the exact same way. If the issue is immutable, it would be nice at least if I could specify a flag such as `--relative-to-output HDMI-A-1` (my left hand screen) for example.

Details on my setup and why it matters to me:
I have a dual monitor setup with resolutions 1920x1200 and 2560x1440, and I have two seperate i3-block-bars configured for each screen. I have a script that gets executed when a certain block is clicked. This script takes the parameters x, y and windowclass. The script positions the window above the window above the button  (each bar passes different coordinate parameters), and toggle/hides it with `move scratchpad|show scratchpad`. It works, but position is off if I click the bar on my left hand screen while working on my right hand screen, and vice versa. I have of course also tested this with running the script manually from terminals on both of my screens as well.

**Comment by emersion** [comment]
Yes, Sway should behave the same as i3 in this case. Wayland doesn't have any limitation which would prevent Sway from doing the same as i3.

**Comment by Xyene** [comment]
There exists `move absolute position x y` that as far as I can tell by reading the code (I don't have a second monitor ATM), should do what you want. Is it what you're looking for?

**Comment by LuyeT** [comment]
Thanks, that works perfectly!

Hopefully, this doesn't come across as nitpicky, I truly respect what you guys have created. Just as a suggestion, it might be advantageous to have "move position x y" behave absolutely as it would in i3, and rename this extra function (which is currently the default) to "move relative position x y". Just in regards to sway being intended as a drop-in replacement for i3.

**Comment by Xyene** [comment]
Happy to hear that works for you.

The option long predates my involvement in this project, so I'll defer the decision of whether to make it i3-compatible to others. I know for some i3-compatibility issues, the divergences were intentional; I'm not sure if that was the case here.

**Comment by 7ff04da4aa6ac37197e249299066ac30** [comment]
Sorry for necroing here, but I have a use case that, I think, should be relevant for at least someone other than me and haven't been able to ascertain a solution.

In my configuration file for sway, I have set Pulse Audio Volume Control to launch as a floating window, resized and moved to an absolute position, as such:

```
for_window [app_id="pavucontrol"] {
	floating enable;
    move position 815 0;
    resize set 544 423
    }
```

This works fine on my main display, which has a certain resolution, meaning that the window stays on the top-right corner of the screen. However, I sometimes use another display, and, upon launching pavucontrol while it is focused, the window is moved to an absolute position, when I'd rather it moved to a relative position, as this other display has a different resolution. In other words, pavucontrol does not stay on the top-right corner as I'd like it to.

Ideally, I'd be able to set something like a relative position:
```
for_window [app_id="pavucontrol"] {
	floating enable;
    move position top right;
    resize set 544 423
    }
```
Is there a way to do so? Or something planned along these lines. I've been reading documentation non-stop, even though this is such a small issue, and swaymsg tells me floating windows can only be moved to absolute positions.

**Comment by bl4ckb0ne** [comment]
This is not the right place to ask a question, please refer to the sway irc channel for those.
